### PR TITLE
clients/stellarcore horizon: Obtain and expose diagnostic events in transaction endpoint

### PR DESCRIPTION
**Pull request by 2opremio**
### What

When submitting failing soroban transactions, Core may return a new `"diagnostic_events"` field with which to debug what happened.

This PR exposes the new field in the Core client. Also, it adds a new extra `"diagnostic_events"` field in Horizon's transaction submission endpoint.

### Why

Improve debugability of failing soroban transactions. Also, it paves to way for exposing diagnostic events in soroban-RPC's `sendTransaction` endpoint.

### Known limitations

Core currently only supports this on nodes with diagnostic events explicitly enabled. @dmkozh is working on enabling the `"diagnostic_events"` field unconditionally.

### CODE REVIEW

File: protocols/stellarcore/tx_response_test.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDiagnosticEventsToSlice(t *testing.T) {
+	events := "AAAAAQAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAgAAAA8AAAAFZXJyb3IAAAAAAAACAAAAAwAAAAUAAAAQAAAAAQAAAAMAAAAOAAAAU3RyYW5zYWN0aW9uIGBzb3JvYmFuRGF0YS5yZXNvdXJjZUZlZWAgaXMgbG93ZXIgdGhhbiB0aGUgYWN0dWFsIFNvcm9iYW4gcmVzb3VyY2UgZmVlAAAAAAUAAAAAAAEJcwAAAAUAAAAAAAG6fA=="
+	slice, err := DiagnosticEventsToSlice(events)
+	assert.NoError(t, err)
+	assert.Len(t, slice, 1)
```
Comment (sreuland) [inline protocols/stellarcore/tx_response_test.go:13]: just for consideration, assert slice element has the expected b64 encoded value of event

File: services/horizon/internal/txsub/errors.go [hunk 2]
```diff
@@ -16,14 +16,15 @@ var (
...
+	ErrBadSequence = &FailedTransactionError{"AAAAAAAAAAD////7AAAAAA==", ""}
 )
 
 // FailedTransactionError represent an error that occurred because
 // stellar-core rejected the transaction.  ResultXDR is a base64
 // encoded TransactionResult struct
 type FailedTransactionError struct {
-	ResultXDR string
+	ResultXDR        string
+	DiagnosticEvents string
```
Comment (sreuland) [inline services/horizon/internal/txsub/errors.go:27]: is it worth following the same naming pattern as`ResultXDR` to name this `DiagnosticEventsXDR` to denote it carries b64 encoded, and/or add another comment in the doc:
```
// DiagnosticEvents is a base64 encoded []xdr.DiagnosticEvent
```

**Review by sreuland — APPROVED** [review]
nice addition!

File: protocols/stellarcore/tx_response.go [hunk 3]
```diff
@@ -25,9 +29,42 @@ type TXResponse struct {
...
+	if err != nil {
+		return nil, err
+	}
+	result := make([]string, len(decoded))
+	for i := 0; i < len(decoded); i++ {
+		encoded, err := xdr.MarshalBase64(decoded[i])
+		if err != nil {
+			return nil, err
+		}
+		result[0] = encoded
```
Comment (Shaptic) [inline protocols/stellarcore/tx_response.go:67]: shouldn't this be `result[i]`?

File: protocols/stellarcore/tx_response_test.go (hunk 1 above)
Comment (Shaptic) [reply protocols/stellarcore/tx_response_test.go:13]: also we should test for more than one to ensure the transform works

File: protocols/stellarcore/tx_response.go (hunk 3 above)
Comment (2opremio) [reply protocols/stellarcore/tx_response.go:67]: Totally! Nice find!!!

File: protocols/stellarcore/tx_response_test.go (hunk 1 above)
Comment (2opremio) [reply protocols/stellarcore/tx_response_test.go:13]: Done

File: protocols/stellarcore/tx_response.go (hunk 3 above)
Comment (2opremio) [reply protocols/stellarcore/tx_response.go:67]: Fixed

File: services/horizon/internal/txsub/errors.go (hunk 2 above)
Comment (2opremio) [reply services/horizon/internal/txsub/errors.go:27]: Done

**Pull request merged**